source/extensions/network/dns_resolver/apple/apple_dns_impl.cc

```diff
@@ -340,7 +359,17 @@ void AppleDnsResolverImpl::PendingResolution::onDNSServiceGetAddrInfoReply(
     }
   }
 
-  if (!(flags & kDNSServiceFlagsMoreComing)) {
+  // If not expecting a v4 query, or the v4 response has been received, consider the v4 response
+  // complete.
+  const bool v4_response_complete =
+      !((query_protocol_ & kDNSServiceProtocol_IPv4) || query_protocol_ == 0) ||
+      pending_response_.v4_response_received_;
```

Comment:
nit: instead of duplicating this non-trivial expression, how 'bout factoring it out into a helper?